src/libsyncengine/requests/serverrequests.cpp

```diff
@@ -1067,63 +1067,41 @@ ExitCode ServerRequests::getPublicLinkUrl(int driveDbId, const QString &fileId,
 
-    if (auto exitCode = job->runSynchronously(); exitCode != ExitCode::Ok) {
+    if (job->runSynchronously() != ExitCode::Ok) {
         if (job->exitCode() == ExitCode::BackError && job->exitCause() == ExitCause::ShareLinkAlreadyExist) {
-            // Get link
-            std::shared_ptr<GetFileLinkJob> job2;
+            // The link already exist, get it
+            job.reset();
             try {
-                job2 = std::make_shared<GetFileLinkJob>(driveDbId, nodeId);
+                job = std::make_shared<GetFileLinkJob>(driveDbId, nodeId);
             } catch (const std::exception &e) {
                 LOG_WARN(Log::instance()->getLogger(),
                          "Error in GetFileLinkJob::GetFileLinkJob for driveDbId=" << driveDbId << " nodeId=" << nodeId.c_str()
                                                                                   << " error=" << e.what());
                 return ExitCode::DataError;
             }
 
-            exitCode = job2->runSynchronously();
-            if (exitCode != ExitCode::Ok) {
+            if (job->runSynchronously() != ExitCode::Ok) {
                 LOG_WARN(Log::instance()->getLogger(),
                          "Error in GetFileLinkJob::GetFileLinkJob for driveDbId=" << driveDbId << " nodeId=" << nodeId.c_str()
-                                                                                  << " code=" << exitCode);
-                return exitCode;
-            }
-
-            Poco::JSON::Object::Ptr resObj = job2->jsonRes();
-            if (!resObj) {
-                LOG_WARN(Log::instance()->getLogger(),
-                         "GetFileLinkJob failed for driveDbId=" << driveDbId << " nodeId=" << nodeId.c_str());
-                return ExitCode::BackError;
-            }
-
-            Poco::JSON::Object::Ptr dataObj = resObj->getObject(dataKey);
-            if (!dataObj) {
-                LOG_WARN(Log::instance()->getLogger(),
-                         "GetFileLinkJob failed for driveDbId=" << driveDbId << " nodeId=" << nodeId.c_str());
-                return ExitCode::BackError;
-            }
-
-            std::string tmp;
-            if (!JsonParserUtility::extractValue(dataObj, urlKey, tmp)) {
-                return ExitCode::BackError;
+                                                                                  << " code=" << job->exitCode());
+                return job->exitCode();
             }
-            linkUrl = QString::fromStdString(tmp);
-
-            return ExitCode::Ok;
         } else {
-            LOG_WARN(Log::instance()->getLogger(), "Error in PostFileLinkJob::PostFileLinkJob for driveDbId="
-                                                           << driveDbId << " nodeId=" << nodeId.c_str() << " code=" << exitCode);
-            return exitCode;
+            LOG_WARN(Log::instance()->getLogger(),
+                     "Error in PostFileLinkJob::PostFileLinkJob for driveDbId=" << driveDbId << " nodeId=" << nodeId.c_str()
```

Comment:
**Suggestion.** Use a lambda to generate the log message. The same message is built at least 3 times.

Reply:
Done in 3479496b5a2452349bf511bb620f292b9214faba